controllers/machinehealthcheck_controller_test.go

```diff
@@ -46,906 +45,847 @@ import (
+					remediated++
 				}
-				return
-			}).Should(Equal(0))
-		})
-
-		Specify("when a Machine has no Node ref for longer than the NodeStartupTimeout", func() {
-			mhc.Spec.ClusterName = cluster.Name
-			mhc.Spec.NodeStartupTimeout = &metav1.Duration{Duration: time.Second}
-			mhc.SetGenerateName("no-noderef-for-longer-than-the-nodestartuptimeout-")
-			Expect(testEnv.Create(ctx, mhc)).To(Succeed())
-
-			// Healthy nodes and machines.
-			fakeNodesMachines(2, true, true)
-			// Unhealthy nodes and machines.
-			fakeNodesMachines(1, false, false)
-			targetMachines := make([]string, len(machines))
-			for i, m := range machines {
-				targetMachines[i] = m.Name
 			}
+			return
+		}).Should(Equal(0))
+	})
 
-			// Make sure the status matches.
-			Eventually(func() *clusterv1.MachineHealthCheckStatus {
-				err := testEnv.Get(ctx, util.ObjectKey(mhc), mhc)
-				if err != nil {
-					fmt.Fprintf(GinkgoWriter, "error retrieving mhc: %s", err)
-					return nil
-				}
-				return &mhc.Status
-			}, timeout).Should(MatchMachineHealthCheckStatus(&clusterv1.MachineHealthCheckStatus{
-				ExpectedMachines:   3,
-				CurrentHealthy:     2,
-				ObservedGeneration: 1,
-				Targets:            targetMachines},
-			))
-
-			// Calculate how many Machines have health check succeeded = false.
-			Eventually(func() (unhealthy int) {
-				machines := &clusterv1.MachineList{}
-				err := testEnv.List(ctx, machines, client.MatchingLabels{
-					"selector": mhc.Spec.Selector.MatchLabels["selector"],
-				})
-				if err != nil {
-					return -1
-				}
+	t.Run("when a Machine has no Node ref for longer than the NodeStartupTimeout", func(t *testing.T) {
+		t.Skip("skipping until made stable")
+		g := NewWithT(t)
+		ctx := context.TODO()
+		cluster := createNamespaceAndCluster(g)
+
+		mhc := newMachineHealthCheck(cluster.Namespace, cluster.Name)
+		mhc.Spec.NodeStartupTimeout = &metav1.Duration{Duration: time.Second}
+
+		g.Expect(testEnv.Create(ctx, mhc)).To(Succeed())
+		defer func(do ...runtime.Object) {
+			g.Expect(testEnv.Cleanup(ctx, do...)).To(Succeed())
+		}(cluster, mhc)
+
+		// Healthy nodes and machines.
+		_, machines, cleanup1 := createMachinesWithNodes(g, 2, true, true, mhc, cluster)
+		defer cleanup1()
+		// Unhealthy nodes and machines.
+		_, unhealthyMachines, cleanup2 := createMachinesWithNodes(g, 1, false, false, mhc, cluster)
+		defer cleanup2()
+		machines = append(machines, unhealthyMachines...)
+
+		targetMachines := make([]string, len(machines))
+		for i, m := range machines {
+			targetMachines[i] = m.Name
+		}
+
+		// Make sure the MHC status matches. We have two healthy machines and
+		// one unhealthy.
+		g.Eventually(func() *clusterv1.MachineHealthCheckStatus {
+			err := testEnv.Get(ctx, util.ObjectKey(mhc), mhc)
+			if err != nil {
+				fmt.Printf("error retrieving mhc: %v", err)
+				return nil
+			}
+			return &mhc.Status
+		}, timeout, time.Second).Should(MatchMachineHealthCheckStatus(&clusterv1.MachineHealthCheckStatus{
```

Comment:
Increase poll frequency? (Will stop commenting, but same comment applies to all one-second frequencies).